bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/RequestLimiter.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.metofficedatahub.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link RequestLimiter} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author David Goodyear - Initial contribution
+ */
+@NonNullByDefault
+public class RequestLimiter {
+
+    public static final int INVALID_REQUEST_ID = -1;
+
+    int requestLimit = 0;
+    int currentRequestCount = 0;
+
+    public int getCurrentRequestCount() {
+        return currentRequestCount;
+    }
+
+    public RequestLimiter() {
+    }
+
+    public void resetLimiter() {
+        resetLimiter(requestLimit);
+    }
+
+    public synchronized void resetLimiter(int newLimit) {
+        requestLimit = newLimit;
+        currentRequestCount = 0;
+    }
+
+    public synchronized void updateLimit(int newLimit) {
+        requestLimit = newLimit;
+    }
+
+    public synchronized int getRequestId() {
```

Comment:
Yeah I can see what you mean, I struggled to think of a better one (late here now as well :) ). I've changed it to getRequestKey() as basically anything calling it is requesting a key that allows the request to be processed. If its passed the limits then the key is invalid. Thank you for all the late nights on my PR's! Hopefully the new name is better? Or possibly it could be token? as its per request?

Reply:
Was more looking towards something down the line of `getRequestCountIfAvailable()`

Reply:
Yeah can go with that as well :) Will update.